pkg/daemon/tasks.go

```diff
@@ -164,6 +169,27 @@ func decodeResult(result interface{}) *runner.Result {
 	return r
 }
 
+type TaskOutcome struct {
+	Outcome task.Outcome `json:"outcome"`
+	Content string       `json:"content"`
+}
+
+func decodeOutcome(t task.Task) TaskOutcome {
+	switch t.Type {
+	case task.TypeBuild:
+		if t.Error == "" {
+			return TaskOutcome{Outcome: task.OutcomeSuccess, Content: fmt.Sprintf("artifacts: %s", t.Result)}
+		}
+		return TaskOutcome{Outcome: task.OutcomeFailure, Content: ""}
+	case task.TypeRun:
+		result := decodeResult(t.Result)
+		return TaskOutcome{Outcome: result.Outcome, Content: result.String()}
+	default:
+		logging.S().Errorw("Unknown task type", "type", t.Type)
+		return TaskOutcome{Outcome: task.OutcomeFailure, Content: ""}
+	}
+}
+
 func unescape(s string) template.HTML {
 	return template.HTML(s)
 }
```

Comment:
(creating a Thread here)

It's a simple fix, it looks like `runner.Result` and the `TaskBuild` type have been added later. Might be worth revisiting how we manage results later.